test/unit/chatAgentRequestHandler.test.ts

```diff
@@ -137,10 +156,19 @@ suite("Chat Agent Request Handler Tests", () => {
                 })(),
             );
 
-        // Mock Language Model API
+        // Had to create a real object instead of using TypeMoq for the response object
```

Comment:
@lewis-sanchez I was not able to get the `mockLanguageModelChatResponse`, looks like it's a broken proxy that cannot be resolved, any idea?
Here's a MVP demmo that can isolate the issue.
```typescript
    test("MVP Demo: TypeMoq proxy issue", async () => {
        // Use the SAME mock object from setup
        await Promise.resolve(mockLanguageModelChatResponse.object);
        expect(true).to.be.true; // Simple assertion
    });

```

As a workaround I had to create an actual object...